apis/supervisor/idp/v1alpha1/types_githubidentityprovider.go.tmpl

```diff
@@ -0,0 +1,218 @@
+	// Defaults to "login:id", which is the user login attribute, followed by a colon, followed by the unique and
+	// unchanging integer ID number attribute. This blends human-readable login names with the unchanging ID value
+	// from GitHub. Note that colons are not allowed in GitHub login attributes or ID numbers, so the colon in the
+	// "login:id" name will always be the login:id separator colon.
+	//
+	// +kubebuilder:default="login:id"
+	// +kubebuilder:validation:Enum={"id","login","login:id"}
+	Username GitHubUsernameClaimSpec `json:"username"`
+
+	// Groups configures which property of the GitHub team record shall determine the group names in Kubernetes.
+	//
+	// Can be either "name" or "slug". Defaults to "slug".
+	//
+	// GitHub team names can contain upper and lower case characters, whitespace, and punctuation (e.g. "Kube admins!").
+	//
+	// GitHub team slugs are lower case alphanumeric characters and may contain dashes (e.g. "kube-admins").
+	//
+	// Either way, group names as presented to Kubernetes will always be prefixed by the GitHub organization
+	// name followed by a forward slash (e.g. "my-org/my-team"). GitHub organization login names can only contain
+	// alphanumeric characters or single hyphens, so the first forward slash `/` will be the separator between
+	// the organization login name and the team name or slug.
+	//
+	// If desired, an admin could configure identity transformation expressions on a FederationDomain to
+	// further customize how these group names are presented to Kubernetes.
+	//
+	// +kubebuilder:default=slug
+	// +kubebuilder:validation:Enum=name;slug
+	Groups GitHubGroupNamesSpec `json:"groups"`
+}
+
+// GitHubClientSpec contains information about the GitHub client that this identity provider will use
+// for web-based login flows.
+type GitHubClientSpec struct {
+	// SecretName contains the name of a namespace-local Secret object that provides the clientID and
+	// clientSecret for an GitHub App or GitHub OAuth2 client.
+	//
+	// This secret must be of type "secrets.pinniped.dev/github-client" with keys "clientID" and "clientSecret".
+	SecretName string `json:"secretName"`
+}
+
+// GitHubIdentityProviderSpec is the spec for configuring an GitHub identity provider.
+type GitHubIdentityProviderSpec struct {
+	// GitHubApi allows configuration for GitHub Enterprise Server
+	//
+	// +optional
+	GitHubApi GitHubApiConfig `json:"github_api,omitempty"`
+
+	// Claims allows customization of the username and groups claims
+	//
+	// +optional
+	Claims GitHubClaims `json:"claims,omitempty"`
+
+	// AllowedOrganizations, when specified, indicates that only users with membership in at least one of the listed
+	// GitHub organizations may log in. In addition, the group membership presented to Kubernetes
+	// will only include teams within the listed GitHub organizations. Additional login rules or group filtering
+	// can optionally be provided as policy expression on any FederationDomain that includes this IDP.
```

Comment:
"on any Pinniped FederationDomain"?

Again just wondering if users will know Pinniped terminology vs GitHub terminology?

Reply:
I've made it more explicit